eslint/eslint-patch/tsconfig.json

```diff
@@ -1,7 +1,8 @@
 {
   "extends": "./node_modules/@rushstack/heft-node-rig/profiles/default/tsconfig-base.json",
-
   "compilerOptions": {
+    "module": "NodeNext",
+    "moduleResolution": "NodeNext",
```

Comment:
Are we sure we want to start using this resolution mode? It's a lot of complexity compared to the resolution most of our code is currently built with.

Reply:
@kevin-y-ang why did you change the resolution mode to be different from the rig's configuration?

Reply:
If my memory serves correctly, I believe I was experimenting with newer versions of @typescript-eslint (because of David's anti-matter ESLint rule idea) which requires a different module resolution: https://github.com/typescript-eslint/typescript-eslint/issues/7284

I will remove it

Reply:
Done

Reply:
Done